src/main/java/seedu/address/ui/SessionListPanel.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.ui;
+
+import java.util.logging.Logger;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.session.Session;
+import seedu.address.model.student.Student;
+
+/**
+ * Panel containing the list of sessions.
+ */
+public class SessionListPanel extends UiPart<Region> {
+    private static final String FXML = "SessionListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(SessionListPanel.class);
+
+    @FXML
+    private ListView<Session> sessionListView;
+
+    /**
+     * Creates a {@code SessionListPanel} with the given {@code ObservableList}.
+     */
+    public SessionListPanel(ObservableList<Student> studentList) {
+        super(FXML);
+        ObservableList<Session> sessionList = FXCollections.observableArrayList();
+        for (Student student : studentList) {
+            sessionList.addAll(student.getListOfSessions());
+        }
+        studentList.addListener((ListChangeListener<Student>) change -> {
+            while (change.next()) {
+                System.out.println("Changes have been made to student");
+                sessionList.clear();
+                for (Student student : change.getList()) {
+                    sessionList.addAll(student.getListOfSessions());
+                }
+            }
+        });
```

Comment:
Do you mind explaining the listener part and what the change.next() does? I a bit catch no ball 😅 . 
Also, which is the part that calls the view to update the session list every time a session is added?

Reply:
The `change.next()` method controls the position of an iterator that goes through the discrete changes in the Student List. Whenever there is a change the listener is called, and the while loop will terminate after all tracked changes are iterated. Here's the [API](https://docs.oracle.com/javase/8/javafx/api/javafx/collections/ListChangeListener.Change.html)

Reply:
I think can imagine the listener as something that's running in the "background" and always monitoring the `ObservableList<Student>`, so whenever there's any changes to the list of students, the code within the block will be invoked. As for the `change.next()`, if there are multiple changes made, they're represented as a 'separate change' (list of changes), so `change.next()` loops through it (might not be 100% correct in this part).

The `ListView<Session>` is actually bound to the `ObservableList<Session> sessionList`, so whenever there's a change in the sessionList, which is shown in this block of code you commented, the ListView will update as well.

_(might not be 100% correct)_

Reply:
Yeah, I believe the "calling" to update the ListView is abstracted away in the implementation of ObservableLists in JavaFx under the Observable collections.